indico/modules/events/timetable/operations.py

```diff
@@ -118,3 +121,57 @@ def fit_session_block_entry(entry, log=True):
         entry.event_new.log(EventLogRealm.management, EventLogKind.change, 'Timetable',
                             "Session block fitted to contents", session.user,
                             data={'Session block': entry.session_block.full_title})
+
+
+def move_timetable_entry(entry, parent=None, day=None):
+    """Move the `entry` to another session or top-level timetable
+
+    :param entry: `TimetableEntry` to be moved
+    :param parent: If specified then the entry will be set as a child
+                         of parent
+    :param day: If specified then the entry will be moved to the
+                        top-level timetable on this day
+    """
+    if bool(parent) + bool(day) != 1:
+        raise Exception("Wrong number of arguments")
+
+    from indico.modules.events.contributions.operations import update_contribution
+
+    updates = {}
+    contrib_update_data = {}
+    if day:
+        new_start_dt = entry.start_dt.replace(day=day.day, month=day.month)
+        updates['start_dt'] = new_start_dt.astimezone(entry.event_new.tzinfo)
+        updates['parent'] = None
+        contrib_update_data = {'session_id': None, 'session_block_id': None}
+    elif parent:
+        children = sorted(parent.children, key=attrgetter('end_dt'), reverse=True)
+        if children:
+            new_start_dt = children[0].end_dt
+        else:
+            new_start_dt = parent.start_dt
+        if (new_start_dt + entry.duration).date() != parent.start_dt.date():
```

Comment:
why do we need to  convert it to event timezone? can't we simply compare utc dates?